DOCCORE-298 Document deprecated operating systems (#3079)
neo4j/docs-operations: modules/ROOT/pages/deprecations.adoc

== Operating systems

Neo4j 2026.05::

> The following distributions are deprecated:

Reviewer: I think you can remove this sentence, otherwise, we need to add a similar one below each version.

Author: But we already have this sentence under 2025.10. I just copied from there. I like this sentence, it makes everything clear

Reviewer: I think we can have it just once then, as well as the sentences on line 188 and 189.

* Ubuntu Server 22.04
* macOS 15 Sequoia
* CentOS Stream 9

> * Windows Server 2022 are deprecated as of 2026.05

Author: ```suggestion
* Windows Server 2022
```

+

> These operating systems will continue to be supported along with the next LTS release in 2026, but new releases after the 2026 LTS will no longer support these platforms.

Reviewer: ```suggestion
These operating system versions will continue to be supported through the next LTS release in 2026, but new releases after 2026 LTS will no longer support them.
```
